modules/net-cloudnat/outputs.tf

```diff
@@ -42,3 +42,12 @@ output "router_name" {
   description = "Cloud NAT router name."
   value       = local.router_name
 }
+
+output "validate_config" {
+  value = null
+
+  precondition {
+    condition     = var.enable_dynamic_port_allocation ? var.enable_endpoint_independent_mapping == false : true
```

Comment:
Never seen this before. Does this work? In any case seems like abusing the language for something you can do with a proper "config" variable with all the required flags.

Reply:
It does work, and in fact is suggested by Terraform devs as a workaround to cross-variable validation not working. But yes, refactored into a config block.